src/LearningOutcomes/LearningOutcomeRouteHandler.spec.ts

```
@@ -1,158 +1,119 @@
 const router = express.Router();
-
-LearningOutcomeRouteHandler.initialize({ router, dataStore });
-
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: true }));
-
 app.use(router);
-
+let dataStore: LearningOutcomeDatastore;
+let driver: MongoDriver;
 const request = supertest(app);
 
-describe('POST /learning-objects/:id/learning-outcomes', () => {
-  it('should return a status of 200 and the id of the inserted outcome', done => {
-    request
-      .post('/learning-objects/someObjectId/learning-outcomes')
-      .send({ outcome: { bloom: 'remember and understand', verb: 'remember' } })
-      .expect('Content-Type', /text/)
-      .expect(200)
-      .then(res => {
-        expect(res.text).toBe('someid');
-        done();
-      });
+describe('LearningOutcomeRouteHandler', () => {
+  beforeAll(async () => {
+    driver = await MongoDriver.build(global['__MONGO_URI__']);
+    dataStore = driver.learningOutcomeStore;
+    LearningOutcomeRouteHandler.initialize({ router, dataStore });
   });
-
-  it('should return a status of 500 and an error message', done => {
-    request
-      .post('/learning-objects/someObjectId/learning-outcomes')
-      .send({ outcome: { bloom: 'bad bloom', verb: 'badverb' } })
-      .expect('Content-Type', /text/)
-      .expect(500)
-      .then(res => {
-        expect(res.text).toMatch('Problem');
-        done();
-      });
+  describe('POST /learning-objects/:id/learning-outcomes', () => {
+    it('should return a status of 200 and the id of the inserted outcome', done => {
+      request
+        .post('/learning-objects/someObjectId/learning-outcomes')
+        .send({ outcome: { bloom: 'remember and understand', verb: 'remember' } })
+        .expect('Content-Type', /text/)
+        .expect(200)
+        .then(res => {
+          expect(res.text).toEqual(expect.any(String));
+          done();
+        });
+    });
+
+    it('should return a status of 500 and an error message', done => {
```

The route should throw a 404 if the requested resource is not found.

---

This route does find the resource though